x-pack/plugins/security_solution/server/lib/telemetry/tasks/endpoint.ts

```diff
@@ -188,7 +190,38 @@ export function createTelemetryEndpointTaskConfig(maxTelemetryBatch: number) {
           )
         : new Map<string, EndpointPolicyResponseDocument>();
 
-      /** STAGE 4 - Create the telemetry log records
+      /** STAGE 4 - Fetch Endpoint Agent Metadata
+       *
+       * Reads Endpoint Agent metadata out of the `.ds-metrics-endpoint.metadata` data stream
+       * and buckets them by Endpoint Agent id and sorts by the top hit. The EP agent will
+       * report its metadata once per day OR every time a policy change has occured. If
+       * a metadata document(s) exists for an EP agent we map to fleet agent and policy
+       */
+      if (endpointData.endpointMetadata === undefined) {
+        logger.debug(`no endpoint metadata to report`);
+        return 0;
```

Comment:
I'm not sure we should short circuit here if there is no metadata - I think we still want to collect the documents. It's possible there is just an issue with that data stream, or for whatever reason, it is temporarily unavailable 

Same with [L211](https://github.com/elastic/kibana/pull/126680/files#diff-49d0c934254fe58a57a7495326d79d4788774b92f309ed847c3547ade6dbc612R211)

The reason why this is done on agents + metrics is that they form the base of the document - if they don't exist we can't build the telemetry document. But we can build it without the metadata if we got this far.

Reply:
thanks for the info and feedback, it makes sense - removed the sort circuit here: https://github.com/elastic/kibana/pull/126680/commits/27503fa9510f05f7864e1e03cbe5cf8fee706c1c